clients/src/main/java/org/apache/kafka/common/requests/CreateTopicsRequest.java

```diff
@@ -0,0 +1,230 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.protocol.ProtoUtils;
+import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.protocol.types.Struct;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CreateTopicsRequest extends AbstractRequest {
+    private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.CREATE_TOPICS.id);
+
+    private static final String REQUESTS_KEY_NAME = "create_topic_requests";
+
+    private static final String TIMEOUT_KEY_NAME = "timeout";
+    private static final String TOPIC_KEY_NAME = "topic";
+    private static final String NUM_PARTITIONS_KEY_NAME = "num_partitions";
+    private static final String REPLICATION_FACTOR_KEY_NAME = "replication_factor";
+    private static final String REPLICA_ASSIGNMENT_KEY_NAME = "replica_assignment";
+    private static final String REPLICA_ASSIGNMENT_PARTITION_ID_KEY_NAME = "partition_id";
+    private static final String REPLICA_ASSIGNMENT_REPLICAS_KEY_NAME = "replicas";
+
+    private static final String CONFIG_KEY_KEY_NAME = "config_key";
+    private static final String CONFIG_VALUE_KEY_NAME = "config_value";
+    private static final String CONFIGS_KEY_NAME = "configs";
+
+    public static final class TopicDetails {
+        public final int numPartitions;
+        public final short replicationFactor;
+        public final Map<Integer, List<Integer>> replicasAssignments;
+        public final Map<String, String> configs;
+
+        private TopicDetails(int numPartitions,
+                             short replicationFactor,
+                             Map<Integer, List<Integer>> replicasAssignments,
+                             Map<String, String> configs) {
+            this.numPartitions = numPartitions;
+            this.replicationFactor = replicationFactor;
+            this.replicasAssignments = replicasAssignments;
+            this.configs = configs;
+        }
+
+        public TopicDetails(int partitions,
+                            short replicationFactor,
+                            Map<String, String> configs) {
+            this(partitions, replicationFactor, new HashMap<Integer, List<Integer>>(), configs);
```

Comment:
Could we use Collections.emptyMap() instead of new HashMap<Integer, List<Integer>>()?